### PR TITLE
Upgrade django, redis, celery, postgres, ...

### LINKED ISSUE #41: Upgrade components
Components to upgrade:
* python
* celery
* postgres (support for 9.6 will end in September 2021!)
etc.

### PR DESCRIPTION
**Pull request by agoncharov-reef**
Fixes #41

### CODE REVIEW

File: {{cookiecutter.repostory_name}}/app/envs/prod/celery-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
```
Comment (mpnowacki-reef) [inline {{cookiecutter.repostory_name}}/app/envs/prod/celery-entrypoint.sh:1]: I think you should quote the original author of this script. What license do they use?

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py [hunk 2]
```diff
@@ -3,16 +3,36 @@
...
-import logging
 from sentry_sdk.integrations.django import DjangoIntegration
 from sentry_sdk.integrations.logging import LoggingIntegration
 
 
 root = environ.Path(__file__) - 2
 
 env = environ.Env(DEBUG=(bool, False))
+
+if env.bool('ENV_FILL_MISSING_VALUES', default=False):
```
Comment (mpnowacki-reef) [inline {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py:23]: some comment here would be most helpful, as this is more complex than code usually found in settings.py

**Review by mpnowacki-reef — CHANGES_REQUESTED** [review]

File: {{cookiecutter.repostory_name}}/app/envs/prod/celery-entrypoint.sh (hunk 1 above)
Comment (agoncharov-reef) [reply {{cookiecutter.repostory_name}}/app/envs/prod/celery-entrypoint.sh:1]: They use "gist.github.com" license :D But you're right, I rewrote the code using only my own head and now I'm the owner >:)

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py (hunk 2 above)
Comment (agoncharov-reef) [reply {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py:29]: Comment added

**Review by mpnowacki-reef — APPROVED** [review]

File: {{cookiecutter.repostory_name}}/app/src/requirements.txt [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
+flower==1.0.0
 {% endif -%}
 {% endif -%}
-psycopg2-binary==2.8.5
-redis==3.5.0
-sentry-sdk==0.14.3
+psycopg2-binary==2.9.1
+redis==3.5.3
+sentry-sdk==1.3.0
+ipython==7.26.0
```
Comment (mlech-reef) [inline {{cookiecutter.repostory_name}}/app/src/requirements.txt:19]: For what?

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py [hunk 4]
```diff
@@ -150,26 +190,28 @@
...
+CELERY_TASK_TIME_LIMIT = 60 * 5
 CELERY_ACCEPT_CONTENT = ['json']
-CELERY_TASK_SERIALIZER = 'json'
+CELERY_SERIALIZER = 'json'
 CELERY_RESULT_SERIALIZER = 'json'
+CELERY_WORKER_PREFETCH_MULTIPLIER = env.int('CELERY_WORKER_PREFETCH_MULTIPLIER', default=10)
+CELERY_BROKER_POOL_LIMIT = env.int('CELERY_BROKER_POOL_LIMIT', default=50)
 {% endif %}
 
+SLACK_BOT_TOKEN = env('SLACK_BOT_TOKEN', default='')
```
Comment (mlech-reef) [inline {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py:214]: Do we use it?

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py [hunk 5]
```diff
@@ -199,13 +241,14 @@
...
-        level=logging.INFO,  # Capture info and above as breadcrumbs
-        event_level=logging.ERROR  # Send error events from log messages
-    )
-
+if SENTRY_DSN := env('SENTRY_DSN', default=''):
     sentry_sdk.init(
-        dsn=env('SENTRY_DSN', default=''),
-        integrations=[DjangoIntegration(), sentry_logging]
+        dsn=SENTRY_DSN,
+        integrations=[
```
Comment (mlech-reef) [inline {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py:247]: https://docs.sentry.io/platforms/python/guides/celery/ ?
If sentry is enabled by default, maybe would be good to solve #42?

File: {{cookiecutter.repostory_name}}/deploy.sh [hunk 6]
```diff
@@ -16,11 +16,16 @@ SERVICES=$(docker-compose ps --services 2>&1 > /dev/stderr \
            | grep -v -e 'is not set' -e nginx -e db -e redis)
 
 docker-compose stop $SERVICES
-docker-compose up -d
 
+# start the app only in order to perform migrations
```
Comment (mlech-reef) [inline {{cookiecutter.repostory_name}}/deploy.sh:20]: Will it fix #44?

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py (hunk 4 above)
Comment (agoncharov-reef) [reply {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py:214]: Not really, removing.

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py (hunk 5 above)
Comment (agoncharov-reef) [reply {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py:247]: Added redis & celery integrations

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py [hunk 7]
```diff
@@ -14,6 +14,8 @@
 import sentry_sdk
 from sentry_sdk.integrations.django import DjangoIntegration
 from sentry_sdk.integrations.logging import LoggingIntegration
+from sentry_sdk.integrations.celery import CeleryIntegration
```
Comment (mlech-reef) [inline {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py:17]: Make it alphabetically ordered

**Review by mlech-reef — APPROVED** [review]

File: {{cookiecutter.repostory_name}}/deploy.sh (hunk 6 above)
Comment (agoncharov-reef) [reply {{cookiecutter.repostory_name}}/deploy.sh]: It should

**Pull request merged**